src/library/aoide/trackcollection.cpp

```diff
@@ -0,0 +1,76 @@
+#include "library/aoide/trackcollection.h"
+
+#include "library/aoide/agent.h"
+#include "library/aoide/libraryfeature.h"
+#include "library/aoide/subsystem.h"
+
+#include "library/asynctrackloader.h"
+#include "library/trackcollection.h"
+
+namespace mixxx {
+
+namespace aoide {
+
+TrackCollection::TrackCollection(
+        UserSettingsPointer userSettings,
+        LegacyTrackCollection* trackCollection,
+        QObject* parent)
+        : ExternalTrackCollection(parent),
+          m_userSettings(userSettings),
+          m_trackCollection(trackCollection),
+          m_trackLoader(new AsyncTrackLoader(trackCollection, this)),
+          m_subsystem(new Subsystem(userSettings, m_trackLoader, this)),
+          m_agent(new Agent(m_subsystem, this)) {
```

Comment:
No. The agent coordinates the selection of an active collection without user interaction. Signals are needed because the whole process is asynchronous.

Reply:
I don't get that from the code.
It looks like as if it is possible to inline all agent code. Is there a worker thread involved? Where is it?

Reply:
I've added a comment to `Agent`. All child components of `TrackCollection` (`Agent`, `Subsystem, `AsyncTrackLoader`) are running on the UI event loop as expected.

The worker thread and all its components are controlled internally by `Subsystem` that acts as a facade. This class also handles startup and shutdown of the child process. The name has been chosen deliberately.